src/components/actions/coreActions.ts

```diff
@@ -0,0 +1,582 @@
+      code: 'trimResponses',
+      path: '../plugins/action_trim_responses/src/index.ts',
+      name: 'Trim duplicate review responses',
+      trigger: 'ON_REVIEW_SUBMIT',
+      event_code: '',
+      sequence: -6,
+      condition: true,
+      parameter_queries: {
+        operator: 'objectProperties',
+        children: ['outputCumulative.reviewStatusHistoryTimestamp'],
+      },
+    },
+    // Update other review status to PENDING or CHANGES REQUESTED after one
+    // review is submitted to another reviewer in the chain of review-levels
+    {
+      code: 'updateReviewStatuses',
+      path: '../plugins/action_update_review_statuses/src/index.ts',
+      name: 'Update Review Statuses',
+      trigger: 'ON_REVIEW_SUBMIT',
+      event_code: null,
+      sequence: -5,
+      condition: true,
+      parameter_queries: {
+        changedResponses: {
+          operator: 'objectProperties',
+          children: ['outputCumulative.updatedResponses'],
+        },
+      },
+    },
+
+    // If sent back to applicant for further information (LOQ), this sets which
+    // review responses are visible to the applicant
+    {
+      code: 'updateReviewVisibility',
+      path: '../plugins/action_update_review_visibility/src/index.ts',
+      name: "Update Applicant's Review Visibility",
+      trigger: 'ON_REVIEW_SUBMIT',
+      event_code: '',
+      sequence: -4,
+      condition: {
+        operator: 'AND',
+        children: [
+          {
+            operator: '=',
+            children: [
+              {
+                operator: 'objectProperties',
+                children: ['applicationData.reviewData.latestDecision.decision'],
+              },
+              'LIST_OF_QUESTIONS',
+            ],
+          },
+          {
+            operator: 'objectProperties',
+            children: ['applicationData.reviewData.isLastLevel'],
+          },
+        ],
+      },
+      parameter_queries: {},
+    },
+    // Will increment stage if and only if last-level decision is "CONFORM". Any
+    // other cases for incrementing stage must be specified in template actions.
+    {
+      code: 'incrementStage',
+      path: '../plugins/action_increment_stage/src/index.ts',
+      name: 'Increment Stage',
+      trigger: 'ON_REVIEW_SUBMIT',
+      event_code: null,
+      sequence: -3,
+      condition: {
+        operator: 'AND',
+        children: [
+          {
+            operator: 'objectProperties',
+            children: ['applicationData.reviewData.isLastLevel'],
+          },
+          {
+            operator: '=',
+            children: [
+              {
+                operator: 'objectProperties',
+                children: ['applicationData.reviewData.latestDecision.decision'],
+              },
+              'CONFORM',
+            ],
+          },
+          // We also want to make sure that it won't move to the next stage if
+          // the reviewer is not reviewing the whole application (NEEDS
+          // RE-WORDING)
```

Comment:
I'll get you to help re-word this better when we go through it together @nmadruga